File: rust/src/leetcode/editor/cn/_0001_two_sum.rs
#[cfg(test)]
mod tests {
    #[allow(unused_imports)]
    use super::*;

    #[test]
    fn test() {
        println!("1、two_sum");
    }
}

#[allow(dead_code)]
struct Solution;

//leetcode submit region begin(Prohibit modification and deletion)
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = std::collections::HashMap::new();
        for (i, &v) in nums.iter().enumerate() {
            if let Some(&j) = map.get(&(target - v)) {
                return vec![j as i32, i as i32];
            }
            map.insert(v, i);
        }
        vec![]
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: go/src/leetcode/editor/cn/0001_two_sum.go
package main

func main() {
	println("1、two_sum")
}

// leetcode submit region begin(Prohibit modification and deletion)
func twoSum(nums []int, target int) []int {
	numberMap := make(map[int]int, 0)

	for i, v := range nums {
		if j, ok := numberMap[target-v]; ok {
			return []int{j, i}
		}
		numberMap[v] = i
	}

	return []int{}
}

//leetcode submit region end(Prohibit modification and deletion)
